examples/example-server-lib/sw_splash.cpp

```diff
@@ -129,48 +358,65 @@ SwSplash::operator std::shared_ptr<SplashSession>() const
     return self;
 }
 
-void SwSplash::operator()(MirConnection* connection)
+void SwSplash::Self::operator()(struct wl_display* display)
 {
-    MirPixelFormat pixel_format = find_8888_format(connection);
+    globals.init(display);
 
-    uint8_t pattern[4] = { 0x14, 0x48, 0xDD, 0xFF };
+    wl_output_add_listener(globals.output, &output_listener, &outputs);
+    wl_display_roundtrip(display);
 
-    switch(pixel_format)
+    for (auto const& oi : outputs)
     {
-    case mir_pixel_format_abgr_8888:
-    case mir_pixel_format_xbgr_8888:
-        std::swap(pattern[2],pattern[0]);
-        break;
+        ctx.width = std::max(ctx.width, oi.second.width);
+        ctx.height = std::max(ctx.height, oi.second.height);
+    }
 
-    case mir_pixel_format_argb_8888:
-    case mir_pixel_format_xrgb_8888:
-        break;
+    struct wl_shm_pool* shm_pool = make_shm_pool(globals.shm, ctx.width * ctx.height * 4, &ctx.content_area);
 
-    default:
-        return;
-    };
+    for (auto& b: ctx.buffers)
+    {
+        b.buffer = wl_shm_pool_create_buffer(shm_pool, 0, ctx.width, ctx.height, ctx.width*4, WL_SHM_FORMAT_ARGB8888);
+        b.available = true;
+        wl_buffer_add_listener(b.buffer, &buffer_listener, &ctx);
+    }
 
+    wl_shm_pool_destroy(shm_pool);
 
-    mir::client::Surface surface{mir_connection_create_render_surface_sync(connection, 42, 42)};
-    MirBufferStream* buffer_stream = mir_render_surface_get_buffer_stream(surface, 42, 42, pixel_format);
+    ctx.display = display;
+    ctx.surface = wl_compositor_create_surface(globals.compositor);
 
-    auto const window = create_window(connection, surface);
+    auto const window = make_scoped(wl_shell_get_shell_surface(globals.shell, ctx.surface), &wl_shell_surface_destroy);
+    wl_shell_surface_set_fullscreen(window.get(), WL_SHELL_SURFACE_FULLSCREEN_METHOD_DEFAULT, 0, nullptr);
+    wl_shell_surface_set_toplevel(window.get());
 
-    MirGraphicsRegion graphics_region;
+    struct wl_callback* first_frame = wl_display_sync(display);
+    wl_callback_add_listener(first_frame, &frame_listener, &ctx);
 
     auto const time_limit = std::chrono::steady_clock::now() + std::chrono::seconds(2);
 
     do
     {
-        mir_buffer_stream_get_graphics_region(buffer_stream, &graphics_region);
+        wl_display_dispatch(display);
 
-        render_pattern(&graphics_region, pattern);
-        mir_buffer_stream_swap_buffers_sync(buffer_stream);
-
-        for (auto& x : pattern)
+        for (auto& x : ctx.pattern)
+        {
             x =  3*x/4;
+        }
 
         std::this_thread::sleep_for(std::chrono::milliseconds(200));
```

Comment:
Hm. Why are we sleeping here?
I guess it's easier than a `poll()`, but `wl_display_dispatch()` will block until there are events available, so it's going to be a bit hit and miss.

Transitioning to `poll()` driven usage can be a followup.